src/test/scala/org/ergoplatform/db/LegacyDbEqualitySpec.scala

```diff
@@ -0,0 +1,98 @@
+package org.ergoplatform.db
+
+import akka.util.ByteString
+import io.iohk.iodb.{ByteArrayWrapper, LSMStore}
+import org.scalatest.{Matchers, PropSpec}
+
+class LegacyDbEqualitySpec extends PropSpec with DBSpec with Matchers {
+
+  private val (keyA, valA) = (byteString32("A"), byteString("1"))
+  private val (keyB, valB) = (byteString32("B"), byteString("2"))
+  private val (keyC, valC) = (byteString32("C"), byteString("3"))
+  private val (keyD, valD) = (byteString32("D"), byteString("4"))
+  private val (keyE, valE) = (byteString32("E"), byteString("5"))
+
+  private val v1 = versionId("v1")
+  private val v2 = versionId("v2")
+  private val v3 = versionId("v3")
+
+  implicit class OptValueOpsBaw(x: Option[Array[Byte]]) {
+    def toBaw: Option[ByteArrayWrapper] = x.map(ByteArrayWrapper.apply)
+  }
+
+  implicit class ValueOpsBaw(x: Array[Byte]) {
+    def toBaw: ByteArrayWrapper = ByteArrayWrapper(x)
+  }
+
+  implicit class KeyValueOpsBaw(xs: Seq[(Array[Byte], Array[Byte])]) {
+    def toBaw: Seq[(ByteArrayWrapper, ByteArrayWrapper)] = xs.map(x => ByteArrayWrapper(x._1) -> ByteArrayWrapper(x._2))
+  }
+
+  private def withDbs(body: (VersionedLDBKVStore, LSMStore) => Unit): Unit =
+    withVersionedStore(100) {
+      body(_, new LSMStore(createTempDir))
+    }
+
+  property("update/read roundtrip equality") {
+    withDbs { (ldb, iodb) =>
+      val toInsert0 = Seq(keyA -> valA, keyB -> valB)
+      val toInsert1 = Seq(keyA -> valC)
+
+      ldb.insert(toInsert0)(v1)
+      iodb.update(v1.toBaw, Seq.empty, toInsert0.toBaw)
+
+      val ldbContents = ldb.get(toInsert0.map(_._1))
+        .collect { case (k, Some(v)) => k -> v }
+        .toBs
+      val iodbContents = iodb.get(toInsert0.toBaw.map(_._1))
+        .collect { case (k, Some(v)) => k -> v }
+        .map { case (k, v) => ByteString(k.data) -> ByteString(v.data) }
+
+      iodbContents should contain theSameElementsAs ldbContents
+
+      ldb.insert(toInsert1)(v2)
+      iodb.update(v2.toBaw, Seq.empty, toInsert1.toBaw)
+
+      val ldbUpdatedValue = ldb.get(keyA)
+      val iodbUpdatedValue = iodb.get(ByteArrayWrapper(keyA))
+
+      ldbUpdatedValue.toBs shouldEqual iodbUpdatedValue.map(x => ByteString(x.data))
+    }
+  }
```

Comment:
Is it possible to have two implementations of the same interface, like for example LdbStore and IodbStore both implement KVStore.
In this case the test will be more convincing, because it will be possible to execute literally the same code of test scenario and compare the results are also the same. 
And this tests will work not only for LdbStore, IodbStore pair, but also for LdbStoreN, LdbStoreN+1 pairs as well.

Reply:
What whould be the point of writing extra code which is assumed to be thrown away later? I suppose the only thing we need to test here is ops data equality.